doc/source/rllib/connector-v2.rst

```diff
@@ -0,0 +1,596 @@
+        .env_runners(env_to_module_connector=lambda env: OneHot())
+    )
+    algo = config.build()
+    # Train one iteration.
+    print(algo.train())
+
+
+ConnectorV2 pipelines in more detail
+------------------------------------
+
+Before diving into more complex and more powerful customization options of the different
+pipeline types (and walking through more examples), the generic inner workings of a
+ConnectorV2 pipeline should be explained in more detail.
+
+
+.. figure:: images/connector_v2/generic_connector_pipeline.svg
+    :width: 1000
+    :align: left
+
+    **Generic ConnectorV2 Pipeline**: All pipelines consist of one or more :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` pieces.
+    When the pipeline is called (for example by an :py:class:`~ray.rllib.env.env_runner.EnvRunner` or a :py:class:`~ray.rllib.core.learner.learner.Learner`),
+    a list of Episodes and the :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` instance are provided.
+    Each :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` in the pipeline takes its predecessor's output
+    (starting on the left side with an initial, empty batch), performs some transformations on the data and passes it on to the next
+    piece in the pipeline. Thereby, all :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` pieces can read from and write into the
+    provided episodes, add any data from these episodes to the batch, or change the data already in the batch.
+    The pipeline then returns the last piece's batch output (right side).
+
+Very importantly, note that the batch output of the pipeline always only lives as long as
+the following :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` forward pass and is discarded right after that.
+However, the list of episodes might survive longer. For example
+if a env-to-module pipeline reads an observation from one episode, changes that observation, and then writes it
+back into the episode, the subsequent module-to-env pipeline is able to see this change.
+The same is true for the Learner pipeline: The episodes that it operates on
+have already been through both env-to-module and module-to-env pipelines and might have been changed by these.
+
+
+Writing more complex custom ConnectorV2s
+----------------------------------------
+
+Besides simply transforming one observation at a time in a preprocessor-style setup as shown preceding,
+here are four more examples showcasing more complex customizations of the different :py:class:`~ray.rllib.connectors.connector_v2.ConnectorV2` pipelines:
+
+* How to add rewards and or actions to the batch.
+* How to stack the N most recent observations.
+* How to manipulate the action logits coming from the :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`.
+
+
+Adding past rewards to the Model's input
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to add the most recent reward to be part of the next observation going into your
```

Comment:
```suggestion
To add the most recent reward to the next observation going into your
```

They're reading this because they want to do it, just tell them how to do it.